c10/util/env.h

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <c10/util/Exception.h>
+#include <cstring>
+#include <iostream>
+#include <sstream>
+
+namespace c10 {
+namespace utils {
+// Reads an environment variable and returns
+// - true,              if set equal to "1"
+// - false,             if set equal to "0"
+// - `default_value`,   otherwise
+//
+// NB:
+// Issues a warning if the value of the environment variable is not 0 or 1.
+bool check_env_bool(const char* name, bool default_value = false) {
```

Comment:
I think this should instead return `c10::optional` (which is our polyfill for [std::optional](https://en.cppreference.com/w/cpp/utility/optional) since we are stuck with C++14 and not C++17). That way the default-arg behavior can be pushed out to clients who could do something like

```cpp
c10::optional<bool> env = check_env("MY_FLAG");
if (env.value_or(false)) {
     ...
}
```